document-portal/document-portal-fuse.c

```diff
@@ -3155,28 +3287,33 @@ xdp_fuse_init (GError **error)
       return FALSE;
     }
 
-  main_ch = fuse_mount (path, &args);
-  if (main_ch == NULL)
+  if (fuse_parse_cmdline (&args, &opts) != 0)
     {
       fuse_opt_free_args (&args);
-      g_set_error (error, XDG_DESKTOP_PORTAL_ERROR, XDG_DESKTOP_PORTAL_ERROR_FAILED,
-                   "Can't mount fuse fs on %s: %s", path, g_strerror (errno));
+      g_set_error (error, XDG_DESKTOP_PORTAL_ERROR,
+                   XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                   "Impossible to parse command line");
       return FALSE;
     }
```

Comment:
Well, I'd prefer to keep the "setup" logic in the calling thread, as I'd see it cleaner.
But in any case, I'd need to send back the error to the caller via thread data.

Reply:
Moving this to the thread would mean you could reduce the amount of data that the fuse thread accesses from the main thread's stack. If you put args, opts, etc on the thread's stack, then there is no possibility that it will become invalid while still in use.

Yes, you'd need to send this error between threads, but (a) you're already doing that, and (b) the function call uses static data, so failures would likely indicate a logic failure. I don't think we gain much in preventing the thread from spawning if this fails.